files/default/slurm/pcluster_slurm_config_generator.py

```diff
@@ -0,0 +1,160 @@
+import errno
+import json
+import logging
+from logging.handlers import RotatingFileHandler
+from os import makedirs, path
+from socket import gethostname
+
+import argparse
+
+from jinja2 import Environment, FileSystemLoader
+
+LOG_FILE_PATH = "/var/log/parallelcluster/pcluster_slurm_config_generator.log"
+
+
+def generate_slurm_config_files(args):
+    """
+    Generate Slurm configuration files.
+
+    For each queue, generate slurm_parallelcluster_{QueueName}_partitions.conf
+    and slurm_parallelcluster_{QueueName}_gres.conf, which contain node info.
+
+    Generate slurm_parallelcluster.conf and slurm_parallelcluster_gres.conf,
+    which includes queue specifc configuration files.
+
+    slurm_parallelcluster.conf is included in main slurm.conf
+    and slurm_parallelcluster_gres.conf is included in gres.conf.
+    """
+    queues_info = _load_queues_info(args.input_file)
+
+    # Make output directories
+    args.output_directory = path.abspath(args.output_directory)
+    pcluster_subdirectory = path.join(args.output_directory, "pcluster")
+    makedirs(pcluster_subdirectory, exist_ok=True)
+    env = _get_jinja_env(args.template_directory)
+
+    # Generate slurm_parallelcluster_{QueueName}_partitions.conf and slurm_parallelcluster_{QueueName}_gres.conf
+    for queue in queues_info["queues_config"]:
+        for file_type in ["partition", "gres"]:
+            logger.info("Generating slurm_parallelcluster_{}_{}.conf".format(queue, file_type))
+            rendered_template = _render_queue_configs(queues_info, queue, file_type, env, pcluster_subdirectory)
+            if not args.dryrun:
+                filename = queues_info["queues_config"][queue]["queue_{}_filename".format(file_type)]
```

Comment:
optional: since we are using python 3.6 to run this scripts we could use f-strings rather than format in case you find it more readable